blocklist-processing-engine/dockerfile

```diff
@@ -1,10 +1,45 @@
-FROM node:12
-WORKDIR /usr/src/app
-COPY . .
-RUN npm i
-# RUN mkdir build
-# RUN mkdir /build/data
-# COPY ./data/BlockedDataNumbers.txt ./build/data/BlockedDataNumbers.txt
-EXPOSE 10000
+FROM --platform=${TARGETPLATFORM:-linux/amd64} ghcr.io/openfaas/of-watchdog:0.8.4 as watchdog
+FROM node:12.16.1-alpine as builder
+
+ARG TARGETPLATFORM
+
+WORKDIR /opt/blocklist-rule-processing-engine
+
+ENV NPM_CONFIG_LOGLEVEL warn
+
+COPY --from=watchdog /fwatchdog /usr/bin/fwatchdog
+RUN chmod +x /usr/bin/fwatchdog
+
+RUN apk --no-cache add curl ca-certificates \
+    && addgroup -S app && adduser -S -g app app
+
+RUN apk add --no-cache -t build-dependencies git make gcc g++ python libtool autoconf automake \
+    && cd $(npm root -g)/npm \
+    && npm config set unsafe-perm true \
+    && npm install -g node-gyp
+
+COPY package.json package-lock.json* /opt/blocklist-rule-processing-engine/
+
+RUN npm ci
```

Comment:
We're running yarn now, so we shouldn't be using npm in the dockerfile. Check [this example](https://github.com/mojaloop/fraud_risk_management/pull/165/files#diff-95e6a1250168c0d85e2a3d8c76757d05aa810f3d0ec42c2a31f5b4215c78c990) where a similar file like this is using yarn

Reply:
+